examples/features/dualstack: Demonstrate Dual Stack functionality

arjan-bal opened the pull request:
This PR add an example to demonstrate the usage of endpoints. The example demonstrates how a channel chooses b/w IPv4 and IPv6 addresses while connecting to backends. This example closely follows the [example in Java](https://github.com/grpc/grpc-java/tree/master/examples/example-dualstack/src/main/java/io/grpc/examples/dualstack).

RELEASE NOTES:
* examples/features/dualstack: New example demonstrating usage of endpoints and dual-stack functionality.

easwars commented:
Wanted to check with you if the example has to be consistent across languages (before starting to review this)?

arjan-bal commented:
> Wanted to check with you if the example has to be consistent across languages (before starting to review this)?

Changed the PR. Added a new example that closely follows Java's example instead of changing the `custombalancer` example. Other languages don't have dualstack examples becaue their resolver API is not public.

easwars commented on examples/features/dualstack/client/main.go:
```diff
@@ -0,0 +1,191 @@
+		// Since the IPv6 address comes first in the resolver list, it will be
+		// given higher priority.
+		fmt.Sprintf("[::1]:%d", port1),
+		// Server 1 is listening only on the IPv4 loopback address.
```
Server 2.

arjan-bal replied:
Fixed the comment.

easwars reviewed (approved):
LGTM modulo minor nits

arjan-bal commented on examples/features/dualstack/README.md:
```diff
@@ -0,0 +1,29 @@
+1.  `127.0.0.1:50050`: Listening only on the IPv4 loopback address.
+1.  `[::1]:50051`: Listening only on the IPv6 loopback address.
+
+The server response will include the include their serving port and address type
```
Removed.

arjan-bal commented on examples/features/dualstack/README.md:
```diff
@@ -0,0 +1,29 @@
+
+The dualstack example uses a custom name resolver that provides both IPv4 and
+IPv6 localhost endpoints for each of 3 server instances. The client will first
+use the default name resolver and load balancers which will only connect tot he
```
Fixed.
